syft/pkg/cataloger/internal/dependency/resolver_test.go

```diff
@@ -101,11 +116,28 @@ func (m *specifierBuilder) WithRequires(p pkg.Package, requires ...string) *spec
 	return m
 }
 
+func (m *specifierBuilder) WithVariant(p pkg.Package, variantName string, requires ...string) *specifierBuilder {
+	if _, ok := m.variants[p.Name]; !ok {
+		m.variants[p.Name] = make(map[string][]string)
+	}
+	m.variants[p.Name][variantName] = append(m.variants[p.Name][variantName], requires...)
+	return m
+}
+
 func (m specifierBuilder) Specifier() Specifier {
 	return func(p pkg.Package) Specification {
```

Comment:
I think this test illustrates where some of my original comment came from.

I don't think I understand instances where the `Specifier` from `specifierBuilder` would want to return []Specification.

Since the argument being given is `p pkg.Package` we should only be getting the values mapped to that package name. 

Is the design you're asking about instead where the `Specifier` returns all `Specifications` irregardless of a `p pkg.Package` being provided as an argument?